toolkit/tools/pkg/imagecustomizerlib/customizeos.go

```diff
@@ -30,47 +30,49 @@ func doOsCustomizations(ctx context.Context, rc *ResolvedConfig, imageConnection
 		return err
 	}
 
-	err = addRemoveAndUpdatePackages(ctx, rc.BuildDirAbs, rc.BaseConfigPath, rc.Config.OS, imageChroot, nil,
```

Comment:
The ordering of these operations is considered part of the API and should not be changed.

Also, there should be a `for` loop for each API.